src/libraries/Microsoft.PowerFx.Json/Functions/TypedParseJSONImpl.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    internal class TypedParseJSONFunctionImpl : TypedParseJSONFunction, IAsyncTexlFunction
+    {
+        public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationToken cancellationToken)
+        {
+            var irContext = IRContext.NotInSource(FormulaType.UntypedObject);
+            var arg0 = args[0];
+
+            if (arg0 is BlankValue || arg0 is ErrorValue)
+            {
+                return arg0;
+            }
+            else if (arg0 is not StringValue)
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = "Runtime type mismatch",
+                    Span = irContext.SourceContext,
+                    Kind = ErrorKind.InvalidArgument
+                });
+            }
+
+            var typeString = (StringValue)args[1];
+            var json = ((StringValue)arg0).Value;
+
+            if (!DType.TryParse(typeString.Value, out DType dtype))
+            {
+                return new ErrorValue(irContext, new ExpressionError()
+                {
+                    Message = $"Internal error: Unable to parse type argument",
```

Comment:
> = $"Internal error: Unable to parse type argument", [](http://example.com/codeflow?start=27&length=52)

Can you add a comment saying that this should never happen, as the typeString will be created by the IR (if that's the case) by serializing the type argument. Same comment for the other two new functions.